devolutions-gateway/src/api/monitoring.rs

```diff
@@ -0,0 +1,222 @@
+use crate::DgwState;
+use crate::http::HttpError;
+use axum::{Json, Router, extract, routing};
+use network_monitor;
+use time::OffsetDateTime;
+
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    let router = Router::new()
+        .route("/config", routing::post(handle_set_monitoring_config))
+        .route("/log/drain", routing::post(handle_drain_log));
+
+    router.with_state(state)
+}
+
+/// Replaces the existing monitoring config with the one provided in the body.
+/// This request will immediately start any new monitors, and will stop
+/// currently active monitors that are no longer in the config.
+///
+/// The configuration is not persisted across restarts.
```

Comment:
suggestion: Here is a wording I find slightly easier to understand. (It’s on me, but I was confused about "will immediately start any new monitors" on first read.

```suggestion
/// Replace the current monitoring configuration with the configuration in the request body.
///
/// Changes take effect immediately:
/// - Starts any monitors newly defined in the payload.
/// - Stops any currently running monitors that are omitted from the payload.
///
/// Note: The configuration is not persisted across process restarts.
```